src/scsiencrypt.h

```diff
@@ -84,6 +88,59 @@ constexpr uint32_t BSLONG(const uint8_t *p)
 #define STENC_BIG_ENDIAN 0
 #endif
 
+enum class encrypt_mode: std::uint8_t {
+  off = 0u,
+  external = 1u,
+  on = 2u,
+};
+
+enum class decrypt_mode: std::uint8_t {
```

Comment:
Would Scott M. suggest auto instead of uint8_t here? Just curious... 
The enum is so much cleaner already. Nice.

Reply:
By default enums are sized as `int` but the underlying field in the SCSI page is a single byte, so it is necessary to require a smaller size.